src/main/java/org/spongepowered/api/command/args/CommandFlags.java

```diff
@@ -486,4 +490,41 @@ public CommandElement buildWith(CommandElement wrapped) {
                     this.unknownLongFlagBehavior, this.anchorFlags);
         }
     }
+
+    private static class FlagElement extends CommandElement {
+
+        @Nullable
+        private final CommandElement valueElement;
+
+        private FlagElement(Text key, @Nullable CommandElement valueElement) {
+            super(key);
+            this.valueElement = valueElement;
+        }
+
+        @Override
+        public void parse(CommandSource source, CommandArgs args, CommandContext context) throws ArgumentParseException {
+            String key = getUntranslatedKey();
+            if (key != null) {
+                context.addFlag(key);
```

Comment:
That's true, I mistook any failures there as throwing an exception that would simply fail but it can still be caught later on. The flag should be added afterwards. The key can't be null either, so that can be taken out as well.